File: f77/build03/drvrinit.f
C
      Subroutine DigSim_Initialization
C
C     This subroutine initializes all of the driver variables to before-use
C     conditions; i.e. counters or references are set to 0, etc.
C
C     Inputs:

C        MAX_NUMBER_OF-PRINTS      - Maximum allowable number of
C                                    communications array locations to
C                                    print [integer]
C        MAX NUMBER_OF_REAL_VALUES - Maximum allowable number of real
C                                    variables to be initialized [Integer]
C
C     Outputs:

C        Number_Of Prints - Number of communications array locations to
C                           print [Integer]
C
C        Number_Of_Real_Values - Current number of real variables to be
C                                initialized [Integer]
C
C        Print_Index           - Communications array location of print data
C                                [Integer Array]
C
C        Set_Real_Value        - Initial values for real variables [Real Array]
C        Set_Real_Value_Index  - Real_Array locations for initial values
C                                [Integer Array]
C
C
C     Internal variables and constants:
C        I - Loop index [Integer]
C
C     Include common block variables and declarations
C
      Include '../driver/global.inc'
      Include '../driver/sysvars.inc'
      Include '../driver/setval.inc'
      Include '../driver/print.inc'
C
C  Exceptions to default type
C
      Integer I
C
C     Initialize initial values common block variables
C
      Number_Of_Real_Values = 0
      Do I = 1, MAX_NUMBER_OF_REAL_VALUES
         Set_Real_Value(I) = 0.0
         Set_Real_Value_Index(I) = 0
      End Do
C
C     Initialize print common block variables
C
      Number Of Prints = 0
      Do I = 1, MAX_NUMBER_OF_PRINTS
         Print_Index(I) = 0
      End Do
C
      Return
      End

File: gcc/build03/drvrinit.c
#include "drvrinit.h"
#include "../driver/global.h"
#include "../driver/sysvars.h"
#include "../driver/setval.h"
#include "../driver/print.h"
void digsim_initialization(void) {
  // This subroutine initializes all of the driver variables to before-use
  // conditions; i.e. counters or references are set to 0, etc.
  // Initialize initial values common block variables
  number_of_real_values = 0;
  for (int i = 0; i < MAX_NUMBER_OF_REAL_VALUES; i++) {
    set_real_value[i].r = 0.0;
    set_real_value_index[i] = 0;
  }
  // Initialize print common block variables
  for (int j = 0; j < MAX_NUMBER_OF_PRINTS; j++) {
    print_index[j] = 0;
  }
}
